tests/queries/0_stateless/02900_issue_55858.sql

```diff
@@ -0,0 +1,10 @@
+set precise_float_parsing = 1;
+
+select cast('2023-01-01' as Float64); -- { serverError 6 }
```

Comment:
It is actually `6 - CANNOT_PARSE_TEXT`
```
SELECT CAST('2023-01-01', 'Float64')

Query id: 38715fd9-d602-4556-9dc6-496b7803b955


0 rows in set. Elapsed: 0.002 sec. 

Received exception from server (version 23.10.1):
Code: 6. DB::Exception: Received from localhost:9001. DB::Exception: Cannot parse string '2023-01-01' as Float64: syntax error at position 4 (parsed just '2023'). Note: there are toFloat64OrZero and toFloat64OrNull functions, which returns zero/NULL instead of throwing exception.: While processing CAST('2023-01-01', 'Float64'). (CANNOT_PARSE_TEXT)

```